mlflow/sklearn/__init__.py

```diff
@@ -455,6 +460,270 @@ def load_model(model_uri):
+         So should be called directly from the "patched method", to ensure it capture
+         correct argument variable name.
+        """
+
+        arg_list = []
+
+        def arg_to_str(arg):
+            if arg is None or np.isscalar(arg):
+                if isinstance(arg, str) and len(arg) > 32:
+                    # truncate too long string
+                    return repr(arg[:32] + '...')
+                return repr(arg)
+            else:
+                # dataset arguments or other non-scalar type argument
+                return _inspect_original_var_name(arg, fallback_name=f'<{arg.__class__.__name__}>')
+
+        for arg in call_pos_args:
+            arg_list.append(arg_to_str(arg))
+
+        for arg_name, arg in call_kwargs.items():
+            arg_list.append(f'{arg_name}={arg_to_str(arg)}')
+
+        arg_list_str = ','.join(arg_list)
+        return f'{call_fn_name}({arg_list_str})'
+
+    def register_metric_info(self, run_id, metric_name, dataset_name, call_command):
+
+        call_cmd_list = self._metric_api_call_info[run_id][metric_name]
+
+        index = len(call_cmd_list)
+        metric_name_with_index = self.gen_name_with_index(metric_name, index)
+        metric_key = f'{metric_name_with_index}_{dataset_name}'
+
+        call_cmd_list.append((metric_name_with_index, call_command))
+
+        self._metric_info_artifact_need_update[run_id] = True
+        return metric_key
+
+    def register_metric_api_call(self, metric_name, call_command, call_pos_args, call_kwargs):
+        """
+        Given a metric api call (include the called metric function, and call arguments)
+        Register the call information (arguments dict) into the `metric_api_call_arg_dict_list_map`
+        and return a tuple of (is_ok, run_id, metric_key)
+        - metric key format is: {eval_dataset_name}_{metric_name}
+          metric_name is generated by metric function name, if there're multiple calls with
+          different call arguments, then appending index to the name.
+        """
+        call_arg_list = list(call_pos_args) + list(call_kwargs.values())
+
+        dataset_id_list = self._pred_result_id_to_dataset_name_and_run_id.keys()
+
+        dataset_name = None
+        run_id = None
+
+        # Note: some metric API the arguments is not like `y_true`, `y_pred`
+        #  e.g.
+        #    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html#sklearn.metrics.roc_auc_score
+        #    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.silhouette_score.html#sklearn.metrics.silhouette_score
+        for arg in call_arg_list:
+            if arg is not None and not np.isscalar(arg) and id(arg) in dataset_id_list:
+                dataset_name, run_id = self._pred_result_id_to_dataset_name_and_run_id[id(arg)]
+                break
+
+        if dataset_name is None or run_id is None:
+            return False, None, None
```

Comment:
```suggestion
        # Note: some metric API the arguments is not like `y_true`, `y_pred`
        #  e.g.
        #    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html#sklearn.metrics.roc_auc_score
        #    https://scikit-learn.org/stable/modules/generated/sklearn.metrics.silhouette_score.html#sklearn.metrics.silhouette_score
        for arg in call_arg_list:
            if arg is not None and not np.isscalar(arg) and id(arg) in dataset_id_list:
                dataset_name, run_id = self._pred_result_id_to_dataset_name_and_run_id[id(arg)]
                break
        else:
            return False, None, None
```

Can we use `if-else` here?